### ISSUE TITLE
Allow sub-site settings on multisite

### ISSUE THREAD

**Issue by bearded-avenger** [AesopInteractive/lasso#76]
add a filter to override mu saving to get_site_option to allow settings to be configured on a per site bases
- [ ] modify instances of is_multisite to `is_plugin_active_for_network`
- [ ] modify logic for saving license key if plugin is network activated
- [ ] modify localized variables if plugin is network activated
- [ ] build network tab and settings if plugin is network activated

**Comment by michaelbeil** [comment]
Noticed this the other day and was gonna make a note. You beat me to it.

**Comment by robmcclel** [comment]
Hey, guys, how's this one coming along?

**Comment by bearded-avenger** [comment]
Hey Rob,
This is in fact pretty deep in it's nature, which is why it hasn't happened yet.

**Comment by robmcclel** [comment]
Thanks, Nick.  Please keep me posted.  I just downloaded the latest release
and am testing now.

**Comment by robmcclel** [comment]
Good news -- the latest release found the structure underneath the our Genji Child Theme and worked as expected for the site admin.  However, it would not work for any other member level, I imagine because there is a setting somewhere in the menu I can't access that needs to be set or adjusted.

When a non-admin tries to create a post, it goes through the motions but results in a page like this:

![lasso bummer](https://cloud.githubusercontent.com/assets/3332660/8238109/7e4791c2-15a8-11e5-891c-141d46b7d4b5.png)

Is there a way I can get around this in the short term?

**Comment by bearded-avenger** [comment]
There are no settings to enable to other users. What user level are you testing? If a use cannot publish_posts then it wont publish a post.

**Comment by robmcclel** [comment]
As the Site admin, I have no problems creating or publishing a post.  Just
did it, in fact.

As a "Contributor" or "Author", Lasso appeared and I was able to start the
post creation process -- but it ended in that "Bummer" page.  I can
understand the Contributor level failing, but I figured "Author" would work
fine.

**Comment by bearded-avenger** [comment]
yep there's a bug with author. but this nothing to do with MU settings. thanks for letting us know.

**Comment by robmcclel** [comment]
Great, thanks!

Again, keep me posted.  I don't want the users to be able to cross-edit
other people's posts, so let me know when this bug is fixed and the plugin
can be used in public.

Feel like it's really getting close.  Pretty excited...

**Comment by bearded-avenger** [comment]
heh, users cannot cross-edit other peoples posts, only users with the correct permissions can do that.

The only bug here is that an author cant publish a post. 

Will update this soon.

**Comment by robmcclel** [comment]
Right -- but as it stands, the only way someone other than the admin could
post is to be an Editor, which means they would have access to other
posts.  For my purposes, users can't be higher than "Author" -- so I'll
wait for the update.

**Comment by michaelbeil** [comment]
@robmcclel to keep up with the author issue see #91.

Also, Lasso respects WordPress' user roles and capabilities as outlined in the codex here: https://codex.wordpress.org/Roles_and_Capabilities. So, no need to worry about cross-editing occurring, unless that user is an admin or editor, or they can edit_others_pages or `edit_others_posts`, etc. 

We will be working to get a fix out for this soon. Thanks for the heads up!

**Comment by robmcclel** [comment]
Thank you, Mike and Nick!  Extremely positive on Lasso here at TS and the
users are definitely looking forward to it.

**Comment by bearded-avenger** [comment]
Awesome Rob!!

**Comment by lkraav** [comment]
Is this issue the one to follow on why lasso doesn't create the correct settings page url when activated on a subsite? From what I can see from 0.9.6 usage of `is_multisite()` the code is prepared only to be network-activated. Not something I'm looking to do.

Documentation confuses me https://lasso.is/help/using-lasso/ because it claims Lasso is subsite-ready.

**Comment by bearded-avenger** [comment]
The documentation says the following (the word "subsite" appears nowhere on that page)

```
Lasso can be network activated, if you want all of the sites on the network to use the same settings. When Lasso is network activated, a settings panel will appear under the Network Settings tab
```

Which is correct, and is how it operates. Per site settings is what this issue report is, and it's on the long list to be implemented. PR's are absolutely welcome if you want to take this on yourself.

**Comment by lkraav** [comment]
My bad, I remembered the wrong link. https://lasso.is/faq/ is the misleading one.

**Comment by bearded-avenger** [comment]
Thanks, I just edited the page. In the beginning per site settings is how it was, but then we needed it to work on a global level for https://story.am (it's what the editor was originally developed for). The best thing to do for now until we get single site support, is make sure the article css class is the same. 

If you have 50 themes and you have 50 sites, then make sure all 50 themes have the same article CSS class so that the editor works on subsites with different themes.

**Comment by robmcclel** [comment]
As a temporary workaround for those wanting to activate on a single site (only) in a MU install, here is something that worked for me.

If you network activate Lasso, it gives you access to the settings.  Fill them out for the one site you want to enable Lasso, then save.  Once saved and you verify it works on your single site, network deactivate the plugin.  Head over to the single site and activate the plugin for that site.  The settings will hold, and Lasso will be active.

Very excited about really getting this plugin out in the wild on the network -- can't wait for 1.0 so I can offer it as an option on all of our sites.
